### PR TITLE
backend/helm: Fix fatal nil-pointer dereferences on invalid chart type install

### LINKED ISSUE #5216: Fatal Nil-Pointer Dereference Server Crash on Unsupported Helm Chart Install (`getChart` returns nil)
### Description

A fatal server crash (nil-pointer dereference) occurs in the background goroutine when a user attempts to install or upgrade a Helm chart that is not of type `"application"` (e.g., `"library"` charts). 

This happens because the `getChart()` function incorrectly returns a `(nil, nil)` result (nil chart, nil error) when the chart type check fails, rather than returning a formatted error. Since the error is `nil`, the caller (`installRelease` or `upgradeRelease`) proceeds to pass the `nil` chart directly into Helm's `Run()` action, which immediately panics and crashes the entire `headlamp-server` process. 

Additionally, this logic passes the `nil` error to the cache with a `"failed"` status. When the UI polls `GET /action/status`, the `GetActionStatus` handler blindly dereferences `*stat.Err`, causing a secondary nil-pointer panic.

### Steps to Reproduce
1. Attempt to install a Helm chart that has `type: library` in its `Chart.yaml` (or any unsupported type) using the Headlamp UI.
2. The installation request is accepted and spawned in a background goroutine.
3. The server immediately panics inside `action.Install.Run()` or `action.Upgrade.Run()` due to the `nil` chart pointer.
4. The entire Headlamp backend process crashes and exits.
5. If the server is restarted and the user queries the action status for that release, the `GetActionStatus` handler will panic due to dereferencing the `nil` error string in the cache.

### Expected Behavior
- Attempting to install an unsupported chart type should safely abort and store a descriptive error in the action cache (e.g., `"chart type 'library' is not installable"`).
- The server must not crash.
- `GetActionStatus` should defensively check for nil errors before dereferencing `*stat.Err`.

### Proposed Fix
1. In `getChart()` (`backend/pkg/helm/release.go`), construct and return an explicit `fmt.Errorf` when the chart type is not `"application"`, instead of returning the stale `nil` error.
2. In `GetActionStatus()`, defensively check `if stat.Err != nil` before dereferencing it.

### PR DESCRIPTION
**Pull request by KumarADITHYA123**
**Summary:** 
Fixes a critical P1 server crash (nil-pointer dereference) that occurs when users attempt to install or upgrade a Helm chart of an unsupported type (such as a `"library"` chart).

 Fixes #5216 

**Changes:**
- **`backend/pkg/helm/release.go` (`getChart`)**: Fixed a semantic bug where `getChart` returned `(nil, nil)` and passed a nil error to `logActionState` upon failing the chart type validation. It now generates and returns an explicit `fmt.Errorf`, preventing the caller (`installRelease` / `upgradeRelease`) from passing a nil chart into the Helm client.
- **`backend/pkg/helm/release.go` (`GetActionStatus`)**: Added a defensive nil-guard before dereferencing `*stat.Err` when `stat.Status == failed`. This prevents HTTP handler panics in edge cases where a failed state was cached without an accompanying error.

**Steps to Test:**
1. Run the backend server (`npm run backend:start`).
2. Attempt to install a Helm chart configured with `type: library` via the UI or by sending a raw `POST` request to `/helm/release/install`.
3. Verify that the server **does not crash**.
4. Verify that the action status properly reports the error: "chart type 'library' is not installable".
5. Run the backend tests (`npm run backend:test`) to confirm no regressions.

**Notes for the Reviewer:**
The core issue was that `loader.Load` succeeded (`err = nil`), but the type validation failed right after. The code then logged and returned the `nil` error, causing the background goroutine to execute `installClient.Run(nil, values)`, which triggers an unrecovered panic in the Helm library.

### CODE REVIEW

File: backend/pkg/helm/release.go [hunk 1]
```diff
@@ -640,8 +641,14 @@ func (h *Handler) getChart(
 
 	// chart is installable only if it is of type application or empty
 	if chart.Metadata.Type != "" && chart.Metadata.Type != "application" {
-		h.logActionState(zlog.Error(), err, actionName, reqChart, reqName, failed, "chart is not installable")
-		return nil, err
+		typeErr := fmt.Errorf(
+			"chart %q has unsupported type %q; only 'application' charts can be installed",
+			reqChart,
+			chart.Metadata.Type,
+		)
```
[the remark this answers is not in the document]
Comment (KumarADITHYA123) [reply backend/pkg/helm/release.go]: Good catch! I've updated the error formatting to exactly match the expected output (chart type 'library' is not installable). I also added TestGetChart_InvalidType in release_internal_test.go to explicitly verify that invalid chart types safely return a non-nil error and update the cache status correctly.

File: backend/pkg/helm/release.go [hunk 2]
```diff
@@ -640,8 +641,14 @@ func (h *Handler) getChart(
...
+		h.logActionState(zlog.Error(), typeErr, actionName, reqChart, reqName, failed, "chart is not installable")
+
+		return nil, typeErr
```
[the remark this answers is not in the document]
Comment (KumarADITHYA123) [reply backend/pkg/helm/release.go:647]: Completely agree on needing regression tests here. I've created release_internal_test.go and added TestGetActionStatus_NilErr. This unit test explicitly simulates a failed state with a nil error pointer, verifying that GetActionStatus safely catches it and returns "unknown error" gracefully without panicking.

**Review by illume — CHANGES_REQUESTED** [review]
Thanks for these changes.


Can you please address the open review comments?

File: backend/pkg/helm/release.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (illume) [reply backend/pkg/helm/release.go:647]: Thanks. Marking as resolved.

File: backend/pkg/helm/release.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (illume) [reply backend/pkg/helm/release.go]: Thanks! also marking as resolved.

File: backend/pkg/helm/release_internal_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	h := &Handler{
+		Cache:       cache.New[interface{}](),
+		EnvSettings: cli.New(),
+	}
+
+	// Create a temp directory for the fake chart
+	chartDir := t.TempDir()
+	chartYaml := filepath.Join(chartDir, "Chart.yaml")
+
+	err := os.WriteFile(chartYaml, []byte("name: test-lib\nversion: 1.0.0\ntype: library\n"), 0600)
```
[the remark this answers is not in the document]
Comment (KumarADITHYA123) [reply backend/pkg/helm/release_internal_test.go]: Great catch! You are completely right—the missing apiVersion: v2 was causing loader.Load() to fail early in the CI pipeline, which bypassed our custom type validation logic.

I've applied the suggested change to include apiVersion: v2 in the dummy Chart.yaml. The test now properly loads the chart and explicitly exercises the non-installable type path as intended. This should also clear the backend CI test failure. Thank you!

**Comment by illume** [comment]
Thanks for that. Looks like it might need a `make backend-format`

**Comment by illume** [comment]
The build-mac CI failure is unrelated. It's a temporary network download issue.

**Comment by KumarADITHYA123** [comment]
@illume The gofumpt formatting has already been applied in the latest force-pushed commit (9d6cee6d). I ran gofumpt -w locally before squashing — the squashed commit includes the fix. The backend test CI is now passing green. Let me know if anything else needs addressing!

**Review by illume — APPROVED** [review]
🎉 thanks!
